internal/pkg/agent/application/gateway/fleet/state.go

```diff
@@ -0,0 +1,22 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fleet
+
+import (
+	"context"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator/state"
+)
+
+// StateFetcher is an interface used to fetch the current state of the coordinator.
```

Comment:
[Nit]
you don't need to say the interface is an interface
```suggestion
// StateFetcher fetches the current state of the coordinator.
```